fdb-relational-grpc/src/test/java/com/apple/foundationdb/relational/jdbc/ResultSetContinuationTest.java

```diff
@@ -0,0 +1,134 @@
+        ResultSet rsProto = TypeConversion.toProtobuf(resultSet);
+
+        try (RelationalResultSetFacade deserializedResultSet = new RelationalResultSetFacade(rsProto)) {
+            List<List<Integer>> rows = toRows(deserializedResultSet, 3);
+            Assertions.assertEquals(List.of(List.of(1, 2, 3), List.of(4, 5, 6)), rows);
+            Continuation convertedContinuation = deserializedResultSet.getContinuation();
+            assertContinuation(continuation, convertedContinuation);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("continuationSource")
+    void testContinuationRoundTrip(Continuation continuation) throws Exception {
+        RelationalResultSet resultSet = TestUtils.resultSet(
+                "TestType",
+                List.of(Types.INTEGER, Types.INTEGER, Types.INTEGER),
+                continuation,
+                TestUtils.row(1, 2, 3), TestUtils.row(4, 5, 6));
+        ResultSet rsProto = TypeConversion.toProtobuf(resultSet);
+
+        try (RelationalResultSetFacade deserializedResultSet = new RelationalResultSetFacade(rsProto)) {
+            // Spend all the rows so we can get the continuation
+            List<List<Integer>> rows = toRows(deserializedResultSet, 3);
+            byte[] returnedContinuation = deserializedResultSet.getContinuation().serialize();
+            byte[] expectedContinuation = continuation.serialize();
+            Assertions.assertArrayEquals(expectedContinuation, returnedContinuation);
+        }
+    }
+
+    @Test
+    void testMidContinuationFails() throws Exception {
+        RelationalResultSet resultSet = TestUtils.resultSet(
+                "TestType",
+                List.of(Types.INTEGER, Types.INTEGER, Types.INTEGER),
+                MockContinuation.BEGIN,
+                TestUtils.row(1, 2, 3), TestUtils.row(4, 5, 6));
+        ResultSet rsProto = TypeConversion.toProtobuf(resultSet);
+
+        try (RelationalResultSetFacade deserializedResultSet = new RelationalResultSetFacade(rsProto)) {
+            Assertions.assertThrows(SQLException.class, () -> deserializedResultSet.getContinuation());
+        }
+    }
+
+    @Test
+    void testResultSetWithNullContinuationFails() throws Exception {
+        RelationalResultSet resultSet = TestUtils.resultSet(
+                "TestType",
+                List.of(Types.INTEGER, Types.INTEGER, Types.INTEGER),
+                null,
+                TestUtils.row(1, 2, 3), TestUtils.row(4, 5, 6));
+        Assertions.assertThrows(NullPointerException.class, () -> TypeConversion.toProtobuf(resultSet));
+    }
+
+    private void assertContinuation(Continuation expected, Continuation actual) {
+        Assertions.assertEquals(expected.atBeginning(), actual.atBeginning());
+        Assertions.assertEquals(expected.atEnd(), actual.atEnd());
+        Assertions.assertEquals(expected.getReason(), actual.getReason());
+        // This, again, assumes that the entire inner continuation is serialized as the state
```

Comment:
Why not call `actual.serialize()`?

Reply:
Actual in this case is the RPC continuation (that wraps around the inner, server-side, continuation). 
I can pull that out of the assert method to the test method, if that helps.

Reply:
In this case the continuation is always `RelationalRpcContinuation` and so they are always the same. Done.